Added a component connecting sentiment analysis

thePetrMarek opened the pull request:
The first version of component connecting sentiment analysis.
The url for Triton inference server is currently defined in `app.local.properties`.

xzayda commented on lib/src/main/kotlin/org/promethist/core/Input.kt:
```diff
@@ -58,6 +58,12 @@ data class Input(
+    val sentimets get() = classes.filter { it.type == Class.Type.Sentiment }
+
+    @get:JsonIgnore
+    val sentiment get() = classes.firstOrNull() ?: error("No sentiment class recognized in input")
```
I prefer to return ?: Class(Class.Type.Sentiment, Sentiment.UNKNOWN.name) to ensure that sentiment sensitive code will work even when sentiment detection not available

thePetrMarek replied:
Error removed, added Class

xzayda commented on runner/app/src/main/kotlin/org/promethist/core/nlp/Sentiment.kt:
```diff
@@ -0,0 +1,54 @@
+        val text = context.input.transcript.text
+        val input = Input(listOf(text))
+        val request = Request(listOf(input))
+        val response = webTarget.path("/sentiment/infer").request().post(Entity.json(request), object : GenericType<Response>() {})
```
add try ... catch to make it work event when embedded service not available (take a look on Cassandra class implementation)

thePetrMarek replied:
Try-catch added

thePetrMarek commented on lib/src/main/kotlin/org/promethist/core/Input.kt:
```diff
@@ -58,6 +58,12 @@ data class Input(
     val intent get() = intents.firstOrNull() ?: error("No intent class recognized in input")
 
+    @get:JsonIgnore
+    val sentimets get() = classes.filter { it.type == Class.Type.Sentiment }
```
Corrected

thePetrMarek commented on runner/app/src/main/kotlin/org/promethist/core/nlp/Sentiment.kt:
```diff
@@ -0,0 +1,54 @@
+
+    private val logger by LoggerDelegate()
+
+    private val idx2Sentiment = mapOf(
```
Added enum class

thePetrMarek commented on runner/app/src/main/kotlin/org/promethist/core/nlp/Sentiment.kt:
```diff
@@ -0,0 +1,54 @@
+package org.promethist.core.nlp
```
Renamed to Triton

thePetrMarek commented:
I addressed your suggestions for improvements. 
I also noticed, that MutableList classes in Input should contain classes sorted by score, otherwise getter `sentiment` returns negative class only (it was added first into the list). Thus, I added sorting of sentiment classes by score into Triton component.

---TESTING---
I created a testing dialogue in development test/test (Stupid name, I know. I created it in hustle. But, just for explanation, my local admin returns error "Name already exists" for any new name I try). I created four paths through dialogue (POSITIVE, NEUTRAL, NEGATIVE, UNKNOWN). I was able to access all paths. I accessed UNKNOWN path by inserting wrong endpoint URL into app.local.properties file. I also printed content of input.sentiments and input.sentiment by logger. Logger's output is correct.
